core/src/main/java/org/infinispan/remoting/rpc/RpcManagerImpl.java

```diff
@@ -134,14 +134,7 @@ private boolean useReplicationQueue(boolean sync) {
          throw new IllegalStateException("Trying to invoke a remote command but the cache is not clustered");
 
       // Set the topology id of the command, in case we don't have it yet
-      if (rpc instanceof TopologyAffectedCommand) {
-         TopologyAffectedCommand topologyAffectedCommand = (TopologyAffectedCommand) rpc;
-         if (topologyAffectedCommand.getTopologyId() == -1) {
-            int currentTopologyId = stateTransferManager.getCacheTopology().getTopologyId();
-            if (trace) log.tracef("Topology id missing on command %s, setting it to %d", rpc, currentTopologyId);
-            topologyAffectedCommand.setTopologyId(currentTopologyId);
-         }
-      }
+      setTopologyId(rpc);
```

Comment:
I'd rather wait for a constant time, expressed in nanos so there's no conversion during the RPC (e.g. `private static final long BIG_DELAY_NANOS = TimeUnit.DAYS.toNanos(1)`).